android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerButtonViewManager.kt

```diff
@@ -241,6 +243,20 @@ class RNGestureHandlerButtonViewManager : ViewGroupManager<ButtonViewGroup>(), R
       _backgroundColor = color
     }
 
+    override fun onInitializeAccessibilityNodeInfo(info: AccessibilityNodeInfo) {
+      super.onInitializeAccessibilityNodeInfo(info)
+
+      // Expose the testID prop as the resource-id name of the view. Black-box E2E/UI testing
+      // frameworks, which interact with the UI through the accessibility framework, do not have
+      // access to view tags. This allows developers/testers to avoid polluting the
+      // content-description with test identifiers.
+      val testId = super.getTag(R.id.react_test_id)
+
+      if (testId is String) {
+        info.setViewIdResourceName(testId)
+      }
```

Comment:
I can see that in `react-native` they also use this method, but with [slightly different check](https://github.com/facebook/react-native/blob/2fee13094b3d384c071978776fd8b7cff0b6530f/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java#L155). Is there a specific reason behind this, or can we do the same?

Reply:
Java vs Kotlin